src/main/scala/beam/agentsim/agents/planning/Tour.scala

```diff
@@ -1,28 +1,38 @@
 package beam.agentsim.agents.planning
 
-import org.matsim.api.core.v01.population.{Activity, PlanElement}
+import org.matsim.api.core.v01.population.PlanElement
 import org.matsim.utils.objectattributes.attributable.Attributes
 
-/***
-  * BEAM
-  */
-class Tour(var trips: Vector[Trip] = Vector()) extends PlanElement {
+class Tour(private var tripsInternal: Vector[Trip] = Vector()) extends PlanElement {
 
-  def originActivity(): Activity = {
-    trips.head.activity
-  }
-
-  def destActivity(): Activity = {
-    trips.reverse.head.activity
-  }
+  def trips: Seq[Trip] = tripsInternal
 
+  // TODO: every time it creates a new attribute. is it supposed to be like that?
   override def getAttributes = new Attributes
 
-  def addTrip(newTrip: Trip): Unit = trips = trips :+ newTrip
+  def addTrip(newTrip: Trip): Unit = tripsInternal = tripsInternal :+ newTrip
 
   def tripIndexOfElement(planElement: PlanElement): Int = {
-    (for (trip <- trips.zipWithIndex
-          if trip._1 == planElement || trip._1.activity == planElement || (trip._1.leg.isDefined && trip._1.leg.get == planElement))
-      yield trip._2).head
+    val indexes = for {
+      tripsWithIndex <- tripsInternal.zipWithIndex
+      if isTripOrActivityPlanElement(tripsWithIndex._1, planElement) || isTripLegPlanElement(
+        tripsWithIndex._1,
+        planElement
+      )
+    } yield tripsWithIndex._2
+    indexes.headOption.getOrElse(-1)
```

Comment:
This is behavioral change. If before it would fail right there, now it will give -1 and then try to understand where it came from.

Reply:
@REASY changed as discussed